roles/openshift_metrics/tasks/generate_cassandra_pvcs.yaml

```diff
@@ -1,17 +1,23 @@
 ---
-- name: Check to see if PVC already exists
+- name: Check to see if PVC already exists in actual namespace
   oc_obj:
     state: list
     kind: pvc
     name: "{{ openshift_metrics_cassandra_pvc_prefix }}-{{ metrics_pvc_index }}"
-    namespace: "{{ openshift_metrics_project }}"
-  register: _metrics_pvc
+    namespace: "{{openshift_metrics_project}}"
+  register: metrics_pvc
+
+# Migration process for PV to another namespace.
+- name: Check if we need to move cassandra PV to another namespace.
+  include_tasks: migrate_cassandra_pvcs.yaml
 
 # _metrics_pvc.results.results | length > 0 returns a false positive
 # so we check for the presence of 'stderr' to determine if the obj exists or not
 # the RC for existing and not existing is both 0
+
+# No PVC exist (either new or old namespace), so it's a new installation
 - when:
-    - _metrics_pvc.results.stderr is defined
+    - metrics_pvc.results.stderr is defined and openshift_metrics_cassandra_old_pvc.results.stderr is defined
```

Comment:
I wonder whether or not we want to check ```metrics_pvc.results.stderr```.  Suppose the playbook does not run to completion. It is aborted after the new PVC is created but before the PV's claim reference is updated. If the playbook is run again, I do not think that the migration process will complete.

Reply:
Since `metrics_pvc` will give us a list, we should be able to see that there are no elements in that list (we may likely need to also check that the object list is defined, if it isn't defined, or if the list is size 0 then we would want to do a new installation).
`openshift_metrics_cassandara_old_pvc` would also be a list

Reply:
@ewolinetz my question is more around whether we should be checking for the existence of the new pvc at all regardless of how that check is implemented. I am a bit concerned that we may lose idempotency with the check for the new pvc. If the installation fails after the new pvc is created but before updating the claim reference (which is the last update we make to the pv), then on subsequent runs of the playbook I do not think we will resume the migration and finish updating the pv. To make it idempontent, I think we should only check for the existence of the old pvc. Thoughts?

Reply:
There were a couple things being discussed in this thread. First, I was raising the question of whether or not we should check at all for the existence of the new PVC here. I fear that by doing so we lose idempotency which could result in the PV's claim reference never getting updated. Given that, let's only check that the old PVC exists.

The second thing discussed is how we check for the (non-)existence of the PVCs. Relying solely on stderr may be a bit of a brittle test. Here is a more robust test which I think is along the lines of what @ewolinetz suggested:

```yaml
- name: Set PVC vars
  set_fact:
    pvc_name: "{{ openshift_metrics_cassandra_pvc_prefix }}-{{ metrics_pvc_index }}"
    pvc_exists: "{{ openshift_metrics_cassandra_old_pvc.results.results[0] | length > 0 }}"

- name: Set fact if we need to move to another namespace
  set_fact:
    openshift_metrics_migrate_namespace: "{{ pvc_exists and openshift_metrics_cassandra_old_pvc.results.results[0]['metadata']['name'] == pvc_name }}"
```
Initially I tried to cram the whole thing into a single expression. I ran into various errors and it was a bit unreadable. Hopefully the checks are pretty clear. First check that there is a PVC. Then check that its name is what we expect it to be.